src/protocol/ipa/mod.rs

```diff
@@ -1137,148 +1139,158 @@ pub mod tests {
-                        IpaQueryConfig::new(
-                            per_user_cap,
-                            MAX_BREAKDOWN_KEY,
-                            ATTRIBUTION_WINDOW_SECONDS,
-                            NUM_MULTI_BITS,
+                    .reconstruct();
+
+                let snapshot = world.metrics_snapshot();
+                assert_baselines(
+                    &format!("semi-honest IPA (cap = {per_user_cap})"),
+                    &snapshot,
+                    [
+                        (RECORDS_SENT, *records_sent.get(&per_user_cap).unwrap()),
+                        (BYTES_SENT, *bytes_sent.get(&per_user_cap).unwrap()),
+                        (
+                            INDEXED_PRSS_GENERATED,
+                            *indexed_prss.get(&per_user_cap).unwrap(),
                         ),
-                    )
-                    .await
-                    .unwrap()
-                })
-                .await;
-
-            let snapshot = world.metrics_snapshot();
-            let (records_baseline, bytes_baseline) = if per_user_cap == 1 {
-                (
-                    RECORDS_SENT_MALICIOUS_BASELINE_CAP_1,
-                    BYTES_SENT_MALICIOUS_BASELINE_CAP_1,
-                )
-            } else {
-                (
-                    RECORDS_SENT_MALICIOUS_BASELINE_CAP_3,
-                    BYTES_SENT_MALICIOUS_BASELINE_CAP_3,
-                )
-            };
-            assert_baselines(
-                &format!("malicious IPA (cap = {per_user_cap})"),
-                &snapshot,
-                [
-                    (RECORDS_SENT, records_baseline),
-                    (BYTES_SENT, bytes_baseline),
-                ],
-            );
+                        (
+                            SEQUENTIAL_PRSS_GENERATED,
+                            *seq_prss.get(&per_user_cap).unwrap(),
+                        ),
+                    ],
+                );
+            }
         }
-    }
 
-    fn assert_baselines<const N: usize>(
-        name: &str,
-        snapshot: &Metrics,
-        baselines: [(&'static str, u64); N],
-    ) {
-        for (metric_name, baseline) in baselines {
-            let actual = snapshot.get_counter(metric_name);
-            assert!(actual <= baseline,
-                    "{metric_name} baseline for {name} has DEGRADED! Expected {baseline}, got {actual}.");
+        #[tokio::test]
+        async fn malicious() {
+            // empirical values per user cap as of Apr 20, 2023.
+            let records_sent: HashMap<u32, u64> = HashMap::from([(1, 36_714), (3, 55_440)]);
+            let bytes_sent: HashMap<u32, u64> = HashMap::from([(1, 137_568), (3, 212_472)]);
+            let indexed_prss: HashMap<u32, u64> = HashMap::from([(1, 76_014), (3, 113_337)]);
+            let seq_prss: HashMap<u32, u64> = HashMap::from([(1, 1108), (3, 1108)]);
+
+            let records = generate_input();
+            for per_user_cap in [1, 3] {
```

Comment:
It might be better to have different tests for the different per-user caps rather than use this construct with `HashMap<u32, u64>` as you do here.  Then you can have a simple expected counts structure.  A function like this would be lovely:

```rust
fn malicious_cap1() {
  malicious(1, ExpectedMetrics {
    records_sent: 36_714,
    bytes_sent: 137_568,
    indexed_prss: 76_014,
    sequential_prss: 1108,
  });
}
```